### PR TITLE
fix-2551: prevented NaN in convoluted matrix

### PR DESCRIPTION
**Pull request by meriki**
## Description
[Tell us about your new feature, improvement, or fix! If relevant, please supplement the PR with images.]

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](http://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

### CODE REVIEW

**Comment by soupault** [comment]
@meriki you need to justify this change by adding a test which will fail on old code and work fine on yours. Also, ensure that your code is PEP-8 compliant (as it mentioned in the PR template).

**Comment by jni** [comment]
@meriki to elaborate on @soupault's point, please look at the tests in `skimage/restoration/tests/test_restoration.py`. Add a function `test_flat_image` there that would have resulted in NaN's in the old code but is fixed in the new code. Thanks!

File: skimage/restoration/deconvolution.py [hunk 1]
```diff
@@ -387,6 +387,7 @@ def richardson_lucy(image, psf, iterations=50, clip=True):
 
     for _ in range(iterations):
         relative_blur = image / convolve_method(im_deconv, psf, 'same')
+        relative_blur[np.isnan(relative_blur)] = 0  # to prevent NaN from propogating throughout convulated matrix
```
Comment (Borda) [inline skimage/restoration/deconvolution.py]: what about `np.nan_to_num()` - https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.nan_to_num.html

Comment (hmaarrfk) [reply skimage/restoration/deconvolution.py]: +1

Comment (stefanv) [reply skimage/restoration/deconvolution.py]: Propagating.

And, "result of convolution" instead of "convulated matrix"?

**Pull request closed without merging**